.github/workflows/build.yml

```diff
@@ -15,16 +17,61 @@ env:
   PLATFORMS: linux/amd64,linux/arm64,linux/arm/v7,linux/arm/v6
 
 jobs:
+  prepare:
+    name: Prepare the Matrix
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Prepare the Matrix
+        id: prepare_matrix
+        shell: bash
+        run: |
+          case ${{ github.event_name }} in
+            pull_request)
+              echo 'tags=[{"tag": "", "ref": "${{ github.ref }}"}]' >> $GITHUB_OUTPUT
+              ;;
+            push)
+              echo 'tags=[{"tag": "${{ github.ref_name }}", "ref": "${{ github.ref }}"}]' >> $GITHUB_OUTPUT
+              ;;
+            schedule)
+              git fetch --all
+              TAG=$(git describe --tags --abbrev=0 || exit 0)
+              if [ -z "$TAG" ]; then
+                echo 'tags=[{"tag": "${{ github.ref_name }}", "ref": "${{ github.ref }}"}]' >> $GITHUB_OUTPUT
+              else
+                echo 'tags=[{"tag": "${{ github.ref_name }}", "ref": "${{ github.ref }}"}, {"tag": "$TAG", "ref": "$(git show-ref --tags -d | grep "/$TAG$" | cut -d ' ' -f 2)"}]' >> $GITHUB_OUTPUT
+              fi
+              ;;
+            *)
+              echo "Run for '${{ github.event_name }}' is not expected"
```

Comment:
If it's not expected, maybe should fail here?

Reply:
Well, 1st of all, the whole workflow will run only if triggered by 1 of the events, defined in the `on` section in line 3, so this is just a fallback section just in case. I let it run with the default ref (also added the default ref_name instead of "" to the tag now) instead of immediately failing the run, as I don't see a negative side of it - if there is something wrong, it will fail at some point later with the real issue. If not, it will run as is with no additional modifications - why not?) I mean, we have to intentionally fail the run only if it might do something dangerous, and I don't see that here. If you do - please share.

Reply:
OK.